src/encode.cpp

```diff
@@ -2613,8 +2617,13 @@ static void encodeReductionLoopBodyAndOutput(
     // t_res[0] = sum(\i. t_input[i / n][i % n] , i < m * n)
 
     // Define this as a splat tensor (num. elems is 1 anyway)
-    t_res = Tensor(t_v.getElemType(), t_v.sum(),
+    // TODO (seongwon): Currently we cover only tensor cases..
+    // TODO: Support memref cases
```

Comment:
I thought linalg.generic for memref was already supported - no?

Reply:
Partially yes but.. to properly support reduction loop, we have to check whether output memref is initialized.